File: ThesisManagementSystem/Models/Thesis.cs
using System;
using System.Collections.Generic;

namespace ThesisManagementSystem.Models
{
    public partial class Thesis
    {
        public int ThesisId { get; set; }
        public string? Title { get; set; }
        public int AuthorId { get; set; }
        public Person? AuthorPerson { get; set; }
        public string? Abstract { get; set; }
        public int? SupervisorId { get; set; }
        public int? CoSupervisorId { get; set; }
        public short ThesisYear { get; set; }
        public string? ThesisType { get; set; }
        public int UniversityId { get; set; }
        public int InstituteId { get; set; }
        public short NumberOfPages { get; set; }
        public string? ThesisLanguage { get; set; }
        public  DateTime SubmissionDate { get; set; }

}
}

File: ThesisManagementSystem/ClientApp/src/app/models/thesis.ts
import { Person } from "./person";

export interface Thesis {
  thesisId: number;
  title: string;
  abstract: string;
  thesisYear: string;
  thesisType: string;
  numberOfPages: string;
  thesisLanguage: string;
  submissionDate: string;
  authorId: number;
  authorPerson: Person;
  supervisorId: number;
  cosupervisorId: number;
  universityId: number;
  instituteId: number;
}
